src/extensions/replay/sessionrecording-utils.ts

```diff
@@ -203,6 +203,13 @@ export class MutationRateLimiter {
         const data = event.data as Partial<mutationCallbackParam>
         const initialMutationCount = this.numberOfChanges(data)
 
+        // do not rate limit events that only remove nodes
+        const removes = data.removes?.length ?? 0
+        const adds = data.adds?.length ?? 0
+        if (removes != 0 && adds === 0) {
```

Comment:
not sure how common is that the removes are always not followed by any adds in the same mutation, but worth giving it a try.